src/cdef.rs

```diff
@@ -404,13 +404,13 @@ pub fn cdef_filter_superblock<T: Pixel>(
             unsafe {
               let xsize = 8 >> xdec;
               let ysize = 8 >> ydec;
-              let dst = out_slice.offset_as_mutable(8 * bx >> xdec, 8 * by >> ydec);
-              let input = in_slice.offset(8 * bx >> xdec, 8 * by >> ydec);
-              assert!(dst.len() >= (ysize - 1) * out_stride + xsize);
-              assert!(input.len() >= ((ysize + 3) * in_stride + xsize + 4) as usize);
-              cdef_filter_block(dst.as_mut_ptr(),
+              assert!(out_slice.rows_iter().len() >= (8 * by >> ydec) + ysize);
```

Comment:
Is there a more direct way to get len() than producing an iterator first?

Reply:
Currently, no, but we can add `rows_count()` (or similar) on `PlaneMutSlice`.

I did not do that because it's only used in `assert!()`.

Reply:
OK, we can skip it.

Reply:
The assert could be a `debug_assert!()` in case.

Reply:
When should we call `assert!()` vs `debug_assert!()` in rav1e?

Here, we call an unsafe code using raw pointers, which is undefined if this assertion fails.

Reply:
Ideally we can move to `debug_assert!()` all the cases in which:
- We want to document the behavior, but
- We have tests to make sure we do not break it
- The code isn't user-facing. When it is, I'd keep the assert and document when it could panic.